wagtail/core/models.py

```diff
@@ -3185,6 +3538,201 @@ def get_comment(self):
         """
         return self.comment
 
+    def log_state_change_action(self, user, action):
+        """Log the approval/rejection action"""
+        page = self.page_revision.as_page_object()
+        next_task = self.workflow_state.get_next_task()
+        next_task_data = None
+        if next_task:
+            next_task_data = {
+                'id': next_task.id,
+                'title': next_task.name
+            }
+        PageLogEntry.objects.log_action(
+            instance=page,
+            action='wagtail.workflow.{}'.format(action),
```

Comment:
Do we still support custom actions on custom task types?

Reply:
yes, `TaskState.approve/reject()` call `TaskState.log_state_change_action()` with approve/reject as the action. Any custom task state should call `log_state_change_action`. One option would be to move this on `TaskState.on_action()`, however this messes up with the order of logged actions. and.. nothing prevents the custom `on_action` to skip the parent on_action entirely. Maybe this needs documenting?